tokenizers/src/models/bpe/model.rs

```diff
@@ -384,28 +362,17 @@ impl Model for BPE {
         self.vocab.len()
     }
 
-    fn tokenize(&self, sentence: Vec<(String, Offsets)>) -> Result<Vec<Token>> {
-        if sentence.is_empty() {
+    fn tokenize(&self, sequence: &str) -> Result<Vec<Token>> {
+        if sequence.is_empty() {
             return Ok(vec![]);
         }
 
-        let mut encoded: Vec<Token> = Vec::with_capacity(sentence.len());
         if self.dropout.is_none() {
-            if let Some(cached_words) = self
-                .cache
-                .as_ref()
-                .and_then(|c| c.get_values(sentence.iter().map(|(s, _)| s)))
-            {
-                return self.tokenize_with_cache(sentence, cached_words);
-            }
-        }
-
-        for (i, (w, initial_offsets)) in sentence.into_iter().enumerate() {
-            let word = self.merge_word(&w)?;
-            encoded.extend(self.word_to_tokens(i as u32, &word, initial_offsets));
+            self.tokenize_with_cache(sequence)
```

Comment:
I think this could be changed such that `tokenize_with_cache` only gets called if no dropout is given and `self.cache` is present, `tokenize_with_cache` could then be passed `&Cache` directly, eliminating some complexity in the method above, something along these lines (untested):

```Rust
fn tokenize_with_cache(&self, sequence: &str, cache: &Cache) -> Result<Vec<Token>> {
    if let Some(ref hit) = cache.get(sequence) {
        [...]
    } else {
        [...]
        cache.set(sequence.to_owned(), word);
    }
}

fn tokenize(&self, sequence, &str) -> Result<Vec<Token>> {
    [...]
    match (self.cache.as_ref(), self.dropout) {
        (Some(cache), Some(_)) => self.tokenize_with_cache(sequence, cache)
        _ => [...]
    }
}
```
```